crates/biome_js_analyze/src/lint/nursery/use_jsx_curly_brace_convention.rs

```diff
@@ -0,0 +1,339 @@
+                            ) = expr
+                            {
+                                Some(node)
+                            } else {
+                                None
+                            }
+                        })
+                    } else {
+                        None
+                    }
+                })?;
+                let jsx_string = make::jsx_string(str_literal.value_token().ok()?);
+                let value = AnyJsxAttributeValue::JsxString(jsx_string);
+                mutation.replace_node(
+                    node.clone(),
+                    make::jsx_attribute_initializer_clause(make::token(T![=]), value),
+                );
+            }
+            (
+                CurlyBraceResolution::RemoveBraces,
+                UseJsxCurlyBraceConventionQuery::AnyJsxChild(node),
+            ) => {
+                if let AnyJsxChild::JsxExpressionChild(expr) = node {
+                    let str_literal = expr.expression().and_then(|expr| {
+                        if let AnyJsExpression::AnyJsLiteralExpression(
+                            AnyJsLiteralExpression::JsStringLiteralExpression(node),
+                        ) = expr
+                        {
+                            Some(node)
+                        } else {
+                            None
+                        }
+                    })?;
+                    let text = &str_literal.value_token().ok()?.token_text();
+                    // trim the quotes off of the string literal
+                    let text_trimmed = text.clone().slice(TextRange::new(
+                        1.into(),
+                        text.len().checked_sub(1.into()).unwrap_or(text.len()),
+                    ));
+                    let jsx_text = biome_js_syntax::AnyJsxChild::JsxText(make::jsx_text(
+                        JsSyntaxToken::new_detached(
+                            JsSyntaxKind::JS_STRING_LITERAL,
+                            &format!("{text_trimmed}"),
+                            [],
+                            [],
+                        ),
+                    ))
+                    .into_syntax()
+                    .into();
+                    mutation.replace_element(node.clone().into_syntax().into(), jsx_text);
+                }
+            }
+        }
+
+        let msg = match state {
+            CurlyBraceResolution::AddBraces => "Add curly braces around the expression.",
+            CurlyBraceResolution::RemoveBraces => "Remove curly braces around the expression.",
+        };
+
+        Some(JsRuleAction::new(
+            ActionCategory::QuickFix,
+            ctx.metadata().applicability(),
+            markup! { {msg} }.to_owned(),
+            mutation,
+        ))
+    }
+}
+
+fn handle_attr_init_clause(
+    attr: &JsxAttributeInitializerClause,
+    has_curly_braces: bool,
+) -> Option<CurlyBraceResolution> {
+    let Ok(node) = attr.value() else {
+        return None;
+    };
```

Comment:
`let node = attr.value().ok()?` is a shorter version.

Reply:
This doesn't work because `attr.value()` returns `Result` and the parent function returns `Option`.

Reply:
But `ok()` transforms it into an `Option`, and then you can use `?`.

Reply:
Ah, i misread that